cmd/protoc-gen-go-grpc/grpc.go

```diff
@@ -309,9 +309,15 @@ func genClientMethod(gen *protogen.Plugin, file *protogen.File, g *protogen.Gene
 		g.P(deprecationComment)
 	}
 	g.P("func (c *", unexport(service.GoName), "Client) ", clientSignature(g, method), "{")
+	g.P("cOpts := make([]", grpcPackage.Ident("CallOption"), ", len(opts))")
+	g.P("for i, opt := range opts {")
+	g.P("	cOpts[i] = opt")
+	g.P("}")
+	g.P("cOpts = append(cOpts, ", grpcPackage.Ident("RegisteredMethod()"), "\"")
```

Comment:
Can be simplified to:

```go
opts := append([]CallOption{grpc.RegisteredMethod()}, opts...) // shadows `opts` so you don't need the subsequent diffs even
``` 

Or use `copy` instead of a loop (and also pre-allocate `len(opts)+1`).

Reply:
Changed to former. Chose that since less lines of code and get rid of subsequent diffs.